Subject:       Control the speaker via environmental variable
Patch author:  Orson Teodoro <orsonteodoro@hotmail.com>

Do not use patch.  It may not work.

Copyright notice for patched files:

/*
 * Copyright (C) 2014-2021 Firejail Authors
 *
 * This file is part of firejail project
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
*/

diff -urp firejail-0.9.64.orig/src/firejail/firejail.h firejail-0.9.64/src/firejail/firejail.h
--- firejail-0.9.64.orig/src/firejail/firejail.h	2020-10-21 04:54:58.000000000 -0700
+++ firejail-0.9.64/src/firejail/firejail.h	2020-10-26 22:15:34.090997999 -0700
@@ -332,6 +332,7 @@ extern int arg_apparmor;	// apparmor
 extern int arg_allow_debuggers;	// allow debuggers
 extern int arg_x11_block;	// block X11
 extern int arg_x11_xorg;	// use X11 security extension
+extern int arg_xpra_speaker;	// enable sound forwarding
 extern int arg_allusers;	// all user home directories visible
 extern int arg_machineid;	// preserve /etc/machine-id
 extern int arg_disable_mnt;	// disable /mnt and /media
Only in firejail-0.9.64/src/firejail: firejail.h.orig
diff -urp firejail-0.9.64.orig/src/firejail/main.c firejail-0.9.64/src/firejail/main.c
--- firejail-0.9.64.orig/src/firejail/main.c	2020-10-21 04:54:58.000000000 -0700
+++ firejail-0.9.64/src/firejail/main.c	2020-10-26 22:15:34.100998393 -0700
@@ -136,6 +136,7 @@ int arg_apparmor = 0;				// apparmor
 int arg_allow_debuggers = 0;			// allow debuggers
 int arg_x11_block = 0;				// block X11
 int arg_x11_xorg = 0;				// use X11 security extension
+int arg_xpra_speaker = 0;			// enable sound forwarding
 int arg_allusers = 0;				// all user home directories visible
 int arg_machineid = 0;				// preserve /etc/machine-id
 int arg_allow_private_blacklist = 0; 		// blacklist things in private directories
@@ -2664,6 +2665,10 @@ int main(int argc, char **argv, char **e
 				exit_err_feature("x11");
 		}
 #endif
+		else if (strcmp(argv[i], "--xpra-speaker=0") == 0)
+			arg_xpra_speaker = 0;
+		else if (strcmp(argv[i], "--xpra-speaker=1") == 0)
+			arg_xpra_speaker = 1;
 		else if (strncmp(argv[i], "--join-or-start=", 16) == 0) {
 			// NOTE: this is second part of option handler,
 			//		 atempt to find and join sandbox is done in other one
Only in firejail-0.9.64/src/firejail: main.c.orig
diff -urp firejail-0.9.64.orig/src/firejail/profile.c firejail-0.9.64/src/firejail/profile.c
--- firejail-0.9.64.orig/src/firejail/profile.c	2020-10-21 04:54:58.000000000 -0700
+++ firejail-0.9.64/src/firejail/profile.c	2020-10-26 22:15:34.100998393 -0700
@@ -1245,6 +1245,14 @@ int profile_check_line(char *ptr, int li
 		return 0;
 	}
 
+	if (strcmp(ptr, "xpra_speaker_off") == 0) {
+		arg_xpra_speaker=0;
+	}
+
+	if (strcmp(ptr, "xpra_speaker_on") == 0) {
+		arg_xpra_speaker=1;
+	}
+
 	// private /etc list of files and directories
 	if (strncmp(ptr, "private-etc ", 12) == 0) {
 		if (arg_writable_etc) {
Only in firejail-0.9.64/src/firejail: profile.c.orig
diff -urp firejail-0.9.64.orig/src/firejail/x11.c firejail-0.9.64/src/firejail/x11.c
--- firejail-0.9.64.orig/src/firejail/x11.c	2020-10-21 04:54:58.000000000 -0700
+++ firejail-0.9.64/src/firejail/x11.c	2020-10-26 22:16:34.473374726 -0700
@@ -808,11 +808,18 @@ static void __attribute__((noreturn)) x1
 	}
 	free(fname);
 
+	char speaker_str[14];
+	if (arg_xpra_speaker) {
+		strcpy(speaker_str,"--speaker=on");
+	} else {
+		strcpy(speaker_str,"--speaker=off");
+	}
+
 	// build attach command
 
 	char * title_arg_str = get_title_arg_str();
 
-	char *attach_argv[] = { "xpra", title_arg_str, "attach", display_str, NULL };
+	char *attach_argv[] = { "xpra", title_arg_str, "attach", speaker_str, display_str, NULL };
 
 	// run attach command
 	client = fork();
@@ -1037,6 +1044,15 @@ static void __attribute__((noreturn)) x1
 			errExit("open");
 	}
 
+	char speaker_str[2];
+	if (arg_xpra_speaker) {
+		strcpy(speaker_str,"1");
+	} else {
+		strcpy(speaker_str,"0");
+	}
+
+	setenv("XPRA_SPEAKER", speaker_str, 1);
+
 	// start
 	server = fork();
 	if (server < 0)
Only in firejail-0.9.64/src/firejail: x11.c.orig
Only in firejail-0.9.64/src/firejail: x11.c.rej
